WIP/calculator.rb

```diff
@@ -0,0 +1,95 @@
+class Calculate
+
+  def initialize(**args)
+    @accumulate = args[:accumulate] || args[:accumulate] == nil
+    @discounts = {}
+  end
+
+  def add_discount(**args)
+
+    name = args[:name]
+    amount = args[:amount]
+    percentage = args[:percentage]
+
+
+    if amount.is_a?(String) || percentage.is_a?(String) || amount.to_i < 0  
+      raise StandardError.new "Not a valid deal"
+    elsif amount == nil && percentage == nil
+      raise StandardError.new "Not a valid deal"
+    elsif percentage.to_i > 100 || percentage.to_i < 0
+      raise StandardError.new "Not a valid deal"
+    end
+
+    @discounts[name] = {
+      name: name,
+      amount: amount,
+      percentage: percentage
+    }
+
+    true
+  end
+
+  def return_discounts
```

Comment:
We don't need this if we have discounts as `attr_accessor` as mentioned in comment above.